File: problems/137-single-number-II/csharp/137_single_number_II.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        Dictionary<int, int> dict = new Dictionary<int, int>();

        foreach (int n in nums)
            if (dict.ContainsKey(n))
                dict[n]++;
            else
                dict.Add(n, 1);

        foreach (KeyValuePair<int, int> kvp in dict) 
            if (kvp.Value == 1)
                return kvp.Key;
            
        return -1;
    }
}

File: problems/137-single-number-II/python/137_single_number_II.py
from typing import Dict, List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        dict: Dict[int, int] = {}
        for n in nums:
            if n in dict:
                dict[n] += 1
            else:
                dict[n] = 1
        for k, v in dict.items():
            if v == 1:
                return k


def test(nums: List[int]) -> None:
    x: int = Solution().singleNumber(nums)
    print(f"The single number of {nums} is {x}")


if __name__ == "__main__":
    test([2, 2, 3, 2])
    test([0, 1, 0, 1, 0, 1, 99])
